### PR TITLE
feat(entities): add entities-upstreams-targets

### PR DESCRIPTION
**Pull request by sumimakito**
# Summary

Verified with Kong Manager's E2E tests: https://github.com/Kong/kong-admin/actions/runs/5818369572

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

File: packages/entities/entities-upstreams-targets/docs/upstreams-config-card.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+A `@copy:success` event is emitted when a user clicks the `Copy JSON` button and the JSON object is copied successfully.
+
+### Usage example
+
+Please refer to the [sandbox](../sandbox/pages/UpstreamsConfigCardPage.vue). The page is accessible by clicking on the row or `View details` button of an existing Upstream.
+
+## TypeScript interfaces
+
+TypeScript interfaces [are available here](https://github.com/Kong/shared-ui-components/blob/main/packages/entities/entities-upstreams-targets/src/types/upstreams-config.ts) and can be directly imported into your host application. The following type interfaces are available for import:
```
Comment (Leopoldthecoder) [inline packages/entities/entities-upstreams-targets/docs/upstreams-config-card.md:104]:
```suggestion
TypeScript interfaces [are available here](https://github.com/Kong/public-ui-components/blob/main/packages/entities/entities-upstreams-targets/src/types/upstreams-config.ts) and can be directly imported into your host application. The following type interfaces are available for import:
```

Could you search globally and replace `Kong/shared-ui-components` with `Kong/public-ui-components`? There're other `shared-ui-components` lying around.

Comment (sumimakito) [reply packages/entities/entities-upstreams-targets/docs/upstreams-config-card.md]: Links updated. Thanks!

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -102,7 +106,7 @@ importers:
         version: 6.1.1(eslint@8.46.0)
       eslint-plugin-vue:
         specifier: ^9.16.1
-        version: 9.17.0(eslint@8.46.0)
+        version: 9.16.1(eslint@8.46.0)
```
Comment (Leopoldthecoder) [inline pnpm-lock.yaml:109]: It seems this is downgraded.

**Review by Leopoldthecoder — APPROVED** [review]
LGTM

**Pull request merged**